### ISSUE TITLE
Too Large; Data Reduction for Vision-Language Pre-Training

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#218]
# TL; DR

- Coreset-select VL training data (K-means!) and enrich GT captions with learned captioner.

# Motivation

<img width="519" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/ed07c958-6dcf-4d43-baa2-b0029d7d3a98">

- VL training dataset is both redundant and often mis-aligned (image, caption).

# Formulation

<img width="1114" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/c2a2f324-2ad7-4a6d-91b0-936258e08295">

- Learn a image captioner with discrete codebook (initialized from keywords).
- Concat base caption with generated caption for better alignment. (*base captions tend to be loosely grounded to the image semantics, while generated captions are limited in diversity*)
- Embed images into discrete codebooks and K-means cluster them (k=3000 for default).
- Sample certain number of samples from each cluster to construct the coreset.

# Evaluation

<img width="549" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/ccaaaba2-67d2-479f-988d-8ba36d30ff1a">

<img width="1108" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/0a36cb08-24f3-450a-a330-78a41ccd115e">

- Caption refining tends to contribute more than sample selection itself.
- Discrete embedding seems to give performance boost.

<img width="1140" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/afad1c64-47d2-4abb-9133-9be375f3cda9">
<img width="542" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/81ef2c51-96c3-47f4-b2cb-e251c39f03a5">
<img width="542" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/6ebfc3a6-b9e6-4912-9875-9aeb3cf304f7">
<img width="545" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/ea6a23a8-dda3-47a5-8d71-535aecaa0b97">
<img width="551" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/eb8ba038-5d82-42f4-b493-78f5093b03f9">
<img width="1122" alt="image" src="https://github.com/reyllama/paper-reviews/assets/46308407/22e83ba4-e96b-4700-ab57-0a1ea6fe39af">

# Inspiration

- Clustering in the codebook space is effective.
- Mis-alignment can undermine VLP especially under limited data regime (e.g., CC3M). The gain from refinement and selection seems to diminish with data scale (and quality).
